### DISCUSSION TITLE
使用 vc-datasource-custom 组件 自定义面和墙材质会引起报错问题

### DISCUSSION THREAD

**Discussion by heroqyc** [zouyaoji/vue-cesium#639] [General]
![image](https://github.com/user-attachments/assets/731a30d5-c7e5-4e49-9593-093e468c0c73)
![image](https://github.com/user-attachments/assets/66d1b278-e6dd-4821-bbb3-9535508a6702)
自定义图片材质会导致这块代码报错 无法 JSON    正确做法应该是什么？

**Comment by zouyaoji** [comment]
这是因为实例化的对象太复杂了，准确说应该是原型链有循环引用，导致JSON.stringify报错。  请用vue-cesium的简化写法，会帮你实例化的。

![image](https://github.com/user-attachments/assets/ecaeb9a9-c83e-48b8-af47-5b177e660c38)

**Reply by zouyaoji** [reply]
截图是vue-cesium扩展的一些type，官方type看这儿：

https://cesium.com/learn/cesiumjs/ref-doc/Material.html?classFilter=Mater

```js
 material: {
              fabric: {
                type: 'Image',
                uniforms: { image: 'https://zouyaoji.top/vue-cesium/images/textures/arrow.png'}
              }
            }
```

**Comment by heroqyc** [comment]
好的 感谢大佬💖
